### DISCUSSION TITLE
Welcome to py_canoe Discussions!

### DISCUSSION THREAD

**Discussion by chaitu-ycr** [chaitu-ycr/py_canoe#31] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

**Comment by The1Gen** [comment]
Hello,

First of all thank you for your contribution, I find py_canoe really useful.

I do face an issue regarding CANoe Tester Present when using py_canoe.

Upon running a python script that sends some diagnostics message using py_canoe, Tester Present gets activated as specified in the loaded configuration. 
However upon end of script execution, Tester Present will get deactivated even if I didn't stopped the measurement from the script, neither the CANoe application. So CANoe UI remains opened, measurement is still running but Tester Present is deactivated.

Any idea why this is happening and how to avoid it?

As a workaround to keep Tester Present activated after script execution, I placed as last instruction an input(), so script execution remains in hang until user input. 

Thanks!

**Reply by chaitu-ycr** [reply]
I will create one method for enabling and disabling tester present in next release 
I will post here once package is ready.

**Comment by chaitu-ycr** [comment]
tester present control logic implemented in latest release. but it seems there is a limitation like you said (once you come out of program, tester present deactivates).

https://github.com/chaitu-ycr/py_canoe/tree/v2.1.0

https://chaitu-ycr.github.io/py_canoe/999_reference/#src.py_canoe.CANoe.control_tester_present
also you can refer this test case for reference --> https://github.com/chaitu-ycr/py_canoe/blob/v2.1.0/tests/test_canoe.py#L142-L158

**Comment by oobashier** [comment]
Is there any way to stop a particular CAN signal by py_canoe
